symbolic-il2cpp/src/usymlite.rs

```diff
@@ -6,13 +6,116 @@
 //! based on the native instruction pointer does not yet exist.
 
 use std::borrow::Cow;
+use std::error::Error;
 use std::ffi::CStr;
+use std::fmt;
 use std::mem;
 use std::os::raw::c_char;
 use std::ptr;
 
-use anyhow::{Error, Result};
+use thiserror::Error;
 
+/// The error type for [`UsymLiteError`].
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum UsymLiteErrorKind {
+    /// Buffer to usym file is misaligned.
+    MisalignedBuffer,
+    /// The header to the usym file is missing or undersized.
+    BadHeader,
+    /// The magic string in the header is missing or malformed.
+    BadMagic,
+    /// The version string in the usym file's header is missing or malformed.
+    InvalidVersion,
```

Comment:
Nit: all other error cases where some datum is invalid are called `BadSomething`.